### PR TITLE
Detail view update

### PR DESCRIPTION
**Pull request by SatanEnglish**
Assuming that data in the deatil view changes.
e.g. I have some label in a detail view lets say it says "Bob".
I press a button and the label changed lets say to "Jim".

When I scroll the view off the screen it will now save that state.
This version doesn't work if the scrolling is fast e.g. pull the scroll down half the page but will save for the general case where someone is browsing down the list.

### CODE REVIEW

**Comment by SatanEnglish** [comment]
This was in regards to issue #198

File: plugins/slick.rowdetailview.js [hunk 1]
```diff
@@ -156,13 +195,28 @@
         collapseItem(_expandedRows[i]);
       }
     }
-
+	
+    // Saves the current state of the detail veiw
+    function saveDetailView(item)
+    {
+        var view = $("#innerDeatilView_" + item.id);
```
Comment (ghiscoding) [inline plugins/slick.rowdetailview.js]: I assume there's a typo here and this `innerDeatilView_` should be `innerDetailView_`
Also `veiw` should be `view` in the comment and everyone it's used
You probably need to update at every place this variable is used, like couple lines below

**Comment by ghiscoding** [comment]
@SatanEnglish 
is your PR finished? 
and what does it bring improvement to the Row Detail plugin?

**Comment by SatanEnglish** [comment]
Had thought it was finished. (But it's not always saving state)
But It's in a stable state and I can just open a new PR when I manage to get it better.

Improvement is that on slow scrolling e.g. not pulling down or up on scroll bar.
We are now saving all changes to the DetailView as it leaves the screen.
If you want to see what I mean look at the old version and follow steps:

1. Open a panel and right click inspect some text.
2. Change the text.
3. Scroll it far off the top of screen.
4. Scroll it back on and the text will have changed back.

In my case I have tabs and the state you left it in wasn't being kept. (Kept returning to first tab once off screen)

This PR saves the view to our object just before it's removed.
It doesn't seem to degrade the performance. 

I'm hoping some time to find a way to check the rows as they are destroyed in DOM.
Alternative would be to update the Core to have an extra event onBeforeRowDestroy. (Likely faster)

**Comment by ghiscoding** [comment]
@SatanEnglish 
The reason of your problem, I think, is because SlickGrid does a re-rendering of the row when go outside of the viewport by scrolling and come to back to it later. I'm not sure if it helps to do a deep clone of your row that you keep in your local array (with `var deepCopiedObj = $.extend({ }, yourObj);` and then overwrite the row content when you come back in the viewport. 

@6pac 
I merged this PR since it looks ok on my side, I tested it locally and it works as expected.

**Comment by SatanEnglish** [comment]
@ghiscoding @6pac 

Need a better way of handling the reload.

Currently inside  function detailSelectionFormatter(row, cell, value, columnDef, dataContext)
I am calling a notify with a timeout to allow for the formater to finish doing it's job like so.

                setTimeout(function () {
                    // trigger an event after toggling
                    _self.onAfterDetailViewFormatterLoad.notify({
                        "grid": _grid,
                        "item": dataContext
                    }, null, _self);
                }, 50);

Can you guys think of a better way of doing this?

Basically this is called every time you scroll the expanding view back into view.
Purpose of this is so that I can fire off scripts to re-setup js that no longer works once the view leaves the cache. (E.g they are no longer attached as the item left the DOM)

**Pull request merged**